### DISCUSSION TITLE
Boss key implementation "issues" regarding Windows Tray Icon

### DISCUSSION THREAD

**Discussion by rodrigopina360** [ppy/osu#27947] [Q&A]
As a bit of a continuation from [this](https://github.com/ppy/osu/discussions/26846) discussion, I'm in the process of implementing the Boss Key logic, with as closely of a behavior to stable as possible. Currently I'm doing the required work to create a tray icon when the game enters the "Boss Key state", and I'm not sure if it's from my lack of knowledge, but I'm trying to use WinAPI, and the code that I've achieved to make it work is finnicky to say the least.
I'd like to ask the community if anyone has any experience in the matter, and if so if there's a better way of creating a tray icon programatically using WinAPI, or even if there's some nuget or related that makes such implementation easier. Perhaps if some of the core stable devs see this and have the time to give their feedback about the process of creating a tray icon.
Thank you for any help provided, and I'm sorry if this isn't the most correct way of going around this "request".

**Comment by peppy** [comment]
See https://github.com/ppy/osu/issues/6790 / https://github.com/ppy/osu-framework/pull/6085 for some preliminary discussion.

The implementation is likely going to have to use winapi, correct. I would provide the osu!stable code related to this functionality but it relies heavily on .NET forms, which we don't really have access to.

**Reply by rodrigopina360** [reply]
Yeah, I also found forms to be relatively easier, but I assumed (and I guess correctly) that it wouldn't be the right way to go around it. I guess I'll just try refining my current implementation, and most importantly check how/if it should affect other OSes.

**Reply by peppy** [reply]
Adding support only for windows is the best first step.
